src/systemverilog/systemverilog.test.ts

```diff
@@ -0,0 +1,194 @@
+			{ startIndex: 11, type: 'delimiter.parenthesis.sv'},
+			{ startIndex: 12, type: 'keyword.posedge.sv'},
+			{ startIndex: 19, type: ''},
+			{ startIndex: 20, type: 'identifier.sv'},
+			{ startIndex: 23, type: 'delimiter.parenthesis.sv'},
+			{ startIndex: 24, type: ''},
+			{ startIndex: 25, type: 'identifier.sv'},
+			{ startIndex: 28, type: ''},
+			{ startIndex: 29, type: 'delimiter.sv'},
+			{ startIndex: 31, type: ''},
+			{ startIndex: 32, type: 'identifier.sv'},
+			{ startIndex: 35, type: ''},
+			{ startIndex: 36, type: 'delimiter.sv'},
+			{ startIndex: 37, type: ''},
+			{ startIndex: 38, type: 'identifier.sv'},
+			{ startIndex: 43, type: 'delimiter.sv'}
+		]
+	}],
+
+	[{
+		line: 'parameter type t_3 = int;',
+		tokens: [
+			{ startIndex: 0,  type: 'keyword.parameter.sv'},
+			{ startIndex: 9,  type: ''},
+			{ startIndex: 10, type: 'keyword.type.sv'},
+			{ startIndex: 14, type: ''},
+			{ startIndex: 15, type: 'identifier.sv'},
+			{ startIndex: 18, type: ''},
+			{ startIndex: 19, type: 'delimiter.sv'},
+			{ startIndex: 20, type: ''},
+			{ startIndex: 21, type: 'keyword.int.sv'},
+			{ startIndex: 24, type: 'delimiter.sv'}
+		]
+	}],
+
+	[{
+		line: 'typedef union packed {',
+		tokens: [
+			{ startIndex: 0,  type: 'keyword.typedef.sv'},
+			{ startIndex: 7, type: ''},
+			{ startIndex: 8,  type: 'keyword.union.sv'},
+			{ startIndex: 13, type: ''},
+			{ startIndex: 14, type: 'keyword.packed.sv'},
+			{ startIndex: 20, type: ''},
+			{ startIndex: 21, type: 'delimiter.curly.sv'}
+		]
+	}],
+
+	[{
+		line: 'a = 8\'hFF;',
+		tokens: [
+			{ startIndex: 0, type: 'identifier.sv' },
+			{ startIndex: 1, type: ''},
+			{ startIndex: 2, type: 'delimiter.sv'},
+			{ startIndex: 3, type: ''},
+			{ startIndex: 4, type: 'number.sv'},
+			{ startIndex: 5, type: 'number.hex.sv'},
+			{ startIndex: 9, type: 'delimiter.sv'}
+		]
+	}],
+
+	[{
+		line: 'a = 2\'b01;',
+		tokens: [
+			{ startIndex: 0, type: 'identifier.sv' },
+			{ startIndex: 1, type: ''},
+			{ startIndex: 2, type: 'delimiter.sv'},
+			{ startIndex: 3, type: ''},
+			{ startIndex: 4, type: 'number.sv'},
+			{ startIndex: 5, type: 'number.binary.sv'},
+			{ startIndex: 9, type: 'delimiter.sv'}
+		]
+	}],
+
+	[{
+		line: 'a = 6\'o654;',
+		tokens: [
+			{ startIndex: 0, type: 'identifier.sv' },
+			{ startIndex: 1, type: ''},
+			{ startIndex: 2, type: 'delimiter.sv'},
+			{ startIndex: 3, type: ''},
+			{ startIndex: 4, type: 'number.sv'},
+			{ startIndex: 5, type: 'number.octal.sv'},
+			{ startIndex: 10, type: 'delimiter.sv'}
+		]
+	}],
+
+	[{
+		line: 'a = 8\'d98;',
+		tokens: [
+			{ startIndex: 0, type: 'identifier.sv' },
+			{ startIndex: 1, type: ''},
+			{ startIndex: 2, type: 'delimiter.sv'},
+			{ startIndex: 3, type: ''},
+			{ startIndex: 4, type: 'number.sv'},
+			{ startIndex: 9, type: 'delimiter.sv'}
+		]
+	}],
+
+	[{
+		line: '10\'bxxxxxxxxxx',
+		tokens: [
+			{ startIndex: 0, type: 'number.sv'}
+		]
+	}],
+
+	[{
+		line: '1\'bz',
```

Comment:
these tests look great. can we also add a couple more where the number is a part of the line. I found a couple more that would be good tests

things like 
if( my_var[3:0] == 4'b0101)

if( my_var[31:0] === 32'h2aB0_113C )


   typedef enum int {FAST_SIM = 0,
                     RANDOM = 1,
                     NOMINAL = 2,
                     START_UP = 3
                     } clock_plan_e;


      bit [63:0] internal_addr =  pft_slice ? {2'b11, external_addr[19:16], external_addr[11:3]} :
                                                    {2'b11, external_addr[19:17], external_addr[12:3]};

Reply:
okay I will add these :)